main.nf

```diff
@@ -1,278 +1,2514 @@
+        val(hisat2_path) from ch_hisat2_index
+
+    output:
+        file("travis.yml") into ch_ciriquant_yml
+
+    when: 'ciriquant' in tool && 'circrna_discovery' in module
+
+    script:
+    index_prefix = fasta.toString() - ~/.fa/
+    fasta_path = fasta.toRealPath()
+    gtf_path = gtf.toRealPath()
+    """
+    export bwa=`whereis bwa | cut -f2 -d':'`
+    export hisat2=`whereis hisat2 | cut -f2 -d':'`
+    export stringtie=`whereis stringtie | cut -f2 -d':'`
+    export samtools=`whereis samtools | cut -f2 -d':' | awk '{print \$1}'`
+
+    touch travis.yml
+    printf "name: ciriquant\n\
+    tools:\n\
+     bwa: \$bwa\n\
+     hisat2: \$hisat2\n\
+     stringtie: \$stringtie\n\
+     samtools: \$samtools\n\n\
+    reference:\n\
+     fasta: ${fasta_path}\n\
+     gtf: ${gtf_path}\n\
+     bwa_index: ${bwa_path}/${index_prefix}\n\
+     hisat_index: ${hisat2_path}/${index_prefix}" >> travis.yml
+    """
+}
+
+/*
+================================================================================
+                          4. Process Input Data
+================================================================================
+*/
+
+/*
+  STEP 4.1: Check Input files
+*/
+
+if(params.input == null){
+   exit 1, "[nf-core/circrna] error: --input was not supplied! Please check '--help' or documentation for details"
+}
+
+csv_path = null
+if(params.input && (has_extension(params.input, "csv"))) csv_path = params.input
+
+ch_input_sample = Channel.empty()
+if(csv_path){
+
+   csv_file = file(csv_path)
+   println csv_file
+   if(csv_file instanceof List) exit 1, "[nf-core/circrna] error: can only accept one CSV file per run."
+   if(!csv_file.exists()) exit 1, "[nf-core/circrna] error: input CSV file could not be found using the path provided: ${params.input}"
+
+   ch_input_sample = extract_data(csv_path)
+
+}else if(params.input && !has_extension(params.input, "csv")){
+
+   log.info ""
+   log.info "Input data log info:"
+   log.info "No input sample CSV file provided, attempting to read from path instead."
+   log.info "Reading input data from path: ${params.input}\n"
+   log.info ""
+   inputSample = retrieve_input_paths(params.input, params.input_type)
+   ch_input_sample = inputSample
+
+}else exit 1, "[nf-core/circrna] error: --input file(s) not correctly supplied or improperly defined, see '--help' flag or documentation"
+
+/*
+  STEP 4.2: Stage BAM/FASTQ files
+*/
+
+if(params.input_type == 'bam'){
+
+   process bam_to_fq{
+
+        publishDir "${params.outdir}/quality_control/preprocessing/bamtofastq", mode: params.publish_dir_mode
+
+        input:
+            tuple val(base), file(bam) from ch_input_sample
+
+        output:
+            tuple val(base), file('*.fq.gz') into fastq_built
+
+        script:
+        """
+        picard "-Xmx${task.memory.toGiga()}g" \
+        SamToFastq \
+        I=$bam \
+        F=${base}_R1.fq.gz \
+        F2=${base}_R2.fq.gz \
+        VALIDATION_STRINGENCY=LENIENT
```

Comment:
Please indent, also use double `\\` as Nextflow consumes one, so with a single `\` the command will end up on one line with big gaps of spaces (readability nice when using Tower / looking at the reports afterwards).

```suggestion
            SamToFastq \\
            I=$bam \\
            F=${base}_R1.fq.gz \\
            F2=${base}_R2.fq.gz \\
            VALIDATION_STRINGENCY=LENIENT
```